llvm/include/llvm/CodeGen/GlobalISel/MachineIRBuilder.h

```diff
@@ -1063,8 +1063,8 @@ class MachineIRBuilder {
 
   /// Build and insert \p Res = G_BUILD_VECTOR with \p Src replicated to fill
   /// the number of elements
-  MachineInstrBuilder buildSplatVector(const DstOp &Res,
-                                       const SrcOp &Src);
+  MachineInstrBuilder buildBuildVectorSplatVector(const DstOp &Res,
```

Comment:
Maybe `buildSplatBuildVector`. Similar to SelectionDAG's `getSplatBuildVector`

Reply:
`buildConstant` can build scalars and splat vectors. It depends on the `type` that you feed in.

Reply:
Updated to buildSplatBuildVector.

@tschuett I am not sure where buildConstant fits in here. Could you clarify?